core/src/main/java/org/jruby/java/proxies/MapJavaProxy.java

```diff
@@ -265,7 +253,7 @@ public RubyHashEntry internalDeleteEntry(final RubyHashEntry entry) {
             // NOTE: this is here to make maps act similar to Hash-es which allow modifications while
             // iterating (meant from the same thread) ... thus we avoid iterating entrySet() directly
             final Map<Object, Object> map = mapDelegate();
-            final Map.Entry[] entries = map.entrySet().toArray(NULL_MAP_ENTRY);
+            final Map.Entry[] entries = map.entrySet().toArray(new Map.Entry[map.size()]);
```

Comment:
we might have had a AIOoBE here - end using `Entry[0]` array resolved the issue

Reply:
How could it be AIOOBE? The map's size can't be less than zero, and constructing a zero-length array is ok.

Though really it should use the same empty array when it's zero-length...

Reply:
https://github.com/jruby/jruby/commit/40f40fb9fae31578ba8c5b2a026df883ef9df7d7 it was a NPE really when the size is bigger than actual contents (in Rhino's case)